### PR TITLE
feat: add message.quote

### PR DESCRIPTION
**Pull request by Milo123459**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes
- [ x ] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [ x ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by almostSouji** [comment]
As discord themselves are planning to remove that feature in favor of proper reply (which may or may not get proper bot API support at some point) I don't personally see much value in this

File: src/structures/Message.js [hunk 1]
```diff
@@ -565,6 +565,19 @@ class Message extends Base {
         : APIMessage.transformOptions(content, options, { reply: this.member || this.author }),
     );
   }
+  /**
+   * Quotes the message.
+   * @param {StringResolvable} [content=''] The content for the message
```
Comment (Papaia) [inline src/structures/Message.js:570]:
```suggestion
   * @param {string} [content=''] The content for the message
```

File: src/structures/Message.js [hunk 2]
```diff
@@ -565,6 +565,19 @@ class Message extends Base {
...
+   * @returns {Promise<Message|Message[]>}
```
Comment (Papaia) [inline src/structures/Message.js:571]:
```suggestion
   * @returns {Promise<Message>}
```

File: src/structures/Message.js [hunk 3]
```diff
@@ -565,6 +565,19 @@ class Message extends Base {
...
+   * @example
+   * // Quote a message
+   * message.quote('Hey, I\'m a quote!')
```
Comment (Papaia) [inline src/structures/Message.js:574]:
```suggestion
   * message.quote("Hey, I'm a quote!")
```

File: typings/index.d.ts [hunk 4]
```diff
@@ -995,6 +995,7 @@ declare module 'discord.js' {
       content?: StringResolvable,
       options?: MessageOptions | MessageAdditions | (MessageOptions & { split?: false }) | MessageAdditions,
     ): Promise<Message>;
+    public quote(content?: StringResolvable): Promise<Message>;
```
Comment (Papaia) [inline typings/index.d.ts:998]:
```suggestion
    public quote(content?: string): Promise<Message>;
```

**Review by Papaia — CHANGES_REQUESTED** [review]

File: src/structures/Message.js [hunk 5]
```diff
@@ -565,6 +565,19 @@ class Message extends Base {
...
+   *   .then(() => console.log(`Sent a quote to ${message.author.username}`))
+   *   .catch(console.error);
+   */
+  quote(content) {
+    return this.channel.send(`> ${this.content}\n${this.member || this.author} ${content}`);
```
Comment (NotSugden) [inline src/structures/Message.js:579]: you should account for multi-line content

maybe also change the behaviour similar to that of [`Message#reply`](https://github.com/discordjs/discord.js/blob/stable/src/structures/Message.js#L561)

File: typings/index.d.ts (hunk 4 above)
Comment (NotSugden) [reply typings/index.d.ts:998]: this overload also needs to be moved ontop or below all of the reply overloads

File: src/structures/Message.js (hunk 5 above)
Comment (Milo123459) [reply src/structures/Message.js:579]: > As discord themselves are planning to remove that feature in favor of proper reply (which may or may not get proper bot API support at some point) I don't personally see much value in this

This pull request is not required as it's probably going to get removed, so I will go a head and close this

**Pull request closed without merging**